components/ui/input.tsx

```diff
@@ -1,25 +1,48 @@
 import * as React from "react";
-
 import { cn } from "@/lib/utils";
+import { Eye, EyeOff } from "lucide-react";
 
 export interface InputProps
-  extends React.InputHTMLAttributes<HTMLInputElement> {}
+  extends React.InputHTMLAttributes<HTMLInputElement> {
+  isPassword?: boolean;
+}
 
 const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, type, ...props }, ref) => {
+  ({ className, type, isPassword = false, ...props }, ref) => {
+    const [showPassword, setShowPassword] = React.useState(false);
+
+    const togglePasswordVisibility = () => setShowPassword(!showPassword);
+
+    const generalClasses = "text-textColor opacity-50 ";
+
     return (
-      <input
-        type={type}
-        className={cn(
-          "flex h-10 w-full rounded-md border border-inputBorder bg-background px-3 py-2 text-sm text-placeholder outline-none file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-sm placeholder:text-muted-foreground focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50 lg:w-[40rem]",
-          className,
+      <div className="relative">
+        <input
+          type={(isPassword && !showPassword) || !isPassword ? type : "text"}
+          className={cn(
+            "!mt-[0.375rem] flex h-10 w-full rounded-md border border-inputBorder bg-background px-3 py-2 text-sm text-placeholder outline-none file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-sm placeholder:text-muted-foreground focus-visible:outline-none disabled:cursor-not-allowed disabled:opacity-50 sm:w-[70vw] lg:w-[39.5rem]",
```

Comment:
never do `focus-visible:outline-none`, it removes focus indication for keyboard users. You can use you `primary` color here.